cluster.tf

```diff
@@ -17,23 +17,18 @@
 // This file was automatically generated from a template in ./autogen
 
 /******************************************
-  Create regional cluster
+  Create Container Cluster
  *****************************************/
 resource "google_container_cluster" "primary" {
   provider = google
 
-  count           = var.regional ? 1 : 0
   name            = var.name
   description     = var.description
   project         = var.project_id
   resource_labels = var.cluster_resource_labels
 
-  region = var.region
-  node_locations = coalescelist(
-    compact(var.zones),
-    sort(random_shuffle.available_zones.result),
-  )
-
+  location          = local.location
+  node_locations    = local.node_locations
```

Comment:
This is an incorrect change. If var.zones is provided, we should use it (even for regional clusters).

Reply:
I thought I preserve previous logic here, checking

Reply:
@morgante is my understanding correct here, for regional cluster var.zones and available zones was combined ? 
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/ed5812423f9e17d71c65473db06d1c052c6cc5aa/cluster_regional.tf#L32-L35

Reply:
you were right, it broken with my change

Reply:
@morgante I did revert this one, can you check latest for main.tf where local.node_locations defined